rotkehlchen/chain/substrate/manager.py

```diff
@@ -363,15 +403,27 @@ def _get_nodes_call_order(self) -> NodesCallOrder:
         """Return a list of maps between a node and its attributes sorted by
         preference.
 
-        NB: currently preference is based on how close they are to the chain
-        height requested via Subscan API, the higher 'weight_block' the better.
+        Own node always has preference, then are ordered depending on how close
+        they are to the chain height; the higher 'weight_block' the better.
         """
-        return sorted(
-            cast(Iterable, self.available_node_attributes_map.items()),
-            key=lambda item: -item[1].weight_block,
+        nodes_sorted: NodesCallOrder = []
+        node_attributes_map = self.available_node_attributes_map.copy()
```

Comment:
Hmm this is not a good idea. Every time the `_get_nodes_call_order()` youare copying the map and then popping own node from it and putting it in front.

Instead what you should do is move this logic to the moment the own node connects. At that point put it to the top of the sorted list.

Reply:
Agree. To be honest we don't know yet about public substrate nodes behaviour and we don't have too many public samples where to choose :D... just parity.